dash-pipeline/SAI/templates/utils.cpp.j2

```diff
@@ -243,6 +309,62 @@ sai_switch_api_t sai_switch_api_impl = {
 };
 
 
+sai_status_t sai_get_port_attribute(
+        _In_ sai_object_id_t port_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list) {
+
+    fprintf(stderr, "sai_get_port_attribute()\n");
+    int i;
+    sai_attribute_t *attr = attr_list;
+
+    for (i = 0; i < attr_count ; i++, attr++) {
+        switch(attr->id) {
+        case SAI_PORT_ATTR_QOS_NUMBER_OF_QUEUES:
+            attr->value.u32 = DASH_BMV2_CPU_QOS_NUMBER_OF_QUEUES;
+            fprintf(stderr, "  sai_get_port_attribute() [0] attr %d SAI_PORT_ATTR_QOS_NUMBER_OF_QUEUES = %d\n", attr->id, attr->value.u32);
+            return SAI_STATUS_SUCCESS;
+
+        default:
+            fprintf(stderr, "  sai_get_port_attribute() [0] attr %d is NOT SUPPORTED - returning SAI_STATUS_SUCCESS anyway\n", attr->id);
+            return SAI_STATUS_SUCCESS;
+        }
+    }
+
+    return SAI_STATUS_SUCCESS;
+}
+
+sai_port_api_t sai_port_api_impl = {
```

Comment:
Make this var static and you don't need = (void *)0 for each of the fields as they will be zero anyway.

Reply:
Hi @vmytnykx I did make static as you suggested, but left in the explicit zero values as a way to document the available APIs and what is not implemented. (Removing the (void *) cleanup up some warnings though).